examples/graphics/mesh-decals.html

```diff
@@ -0,0 +1,245 @@
+            this.positions[12 * i + 6] = pos.x + size * Math.sin(angle);      this.positions[12 * i + 7] = 0;     this.positions[12 * i + 8] = pos.z + size * Math.cos(angle);      angle += Math.PI * 0.5;
+            this.positions[12 * i + 9] = pos.x + size * Math.sin(angle);      this.positions[12 * i + 10] = 0;    this.positions[12 * i + 11] = pos.z + size * Math.cos(angle);     angle += Math.PI * 0.5;
+        }
+
+        // helper function to update required vertex streams
+        function UpdateMesh(device, mesh, updatePositions, updateColors, initAll) {
+
+            // update positions when needed
+            if (updatePositions)
+                self.mesh.setPositions(self.positions);
+
+            // update colors when needed
+            if (updateColors)
+                self.mesh.setColors32(self.colors);
+
+            // update indices and uvs only one time, as they never change
+            if (initAll) {
+                mesh.setIndices(self.indices);
+                mesh.setUvs(0, self.uvs);
+            }
+
+            mesh.updateGeometry(device, pc.PRIMITIVE_TRIANGLES);
+        }
+
+        // Create a mesh with dynamic vertex buffer and static index buffer
+        this.mesh = new pc.Mesh();
+        this.mesh.clear(true, false);
+        UpdateMesh(this.app.graphicsDevice, this.mesh, true, true, true);
+
+        // create material
+        this.material = new pc.StandardMaterial();
+        this.material.useLighting = false;      // turn off lighting - we use emissive texture only. Also, lighting needs normal maps which we don't generate
+        this.material.diffuse = new pc.Color(0, 0, 0);
+        this.material.emissiveVertexColor = true;
+        this.material.blendType = pc.BLEND_ADDITIVE;     // additive alpha blend
+        this.material.depthWrite = false;        // optimization - no need to write to depth buffer, as decals are part of the ground plane
+
+        // Create the mesh instance
+        var node = new pc.GraphNode();
+        var meshInstance = new pc.MeshInstance(node, this.mesh, this.material);
+
+        // Create a model and add the mesh instance to it
+        var model = new pc.Model();
+        model.graph = node;
+        model.meshInstances = [ meshInstance ];
+
+        // Create Entity and add it to the scene
+        this.entity = new pc.Entity();
+        app.root.addChild(this.entity);
+
+        // Add a model compoonent
+        app.systems.model.addComponent(this.entity, {
+            type: 'asset',
+            castShadows: false
+        });
+        this.entity.model.model = model;
+
+        // load a texture, create material and apply it to our MeshInstance
+        app.assets.loadFromUrl("../assets/textures/spark.png", "texture", function (err, asset) {
+            // update material with texture
+            this.entity.model.model.meshInstances[0].material.emissiveMap = asset.resource;
```

Comment:
This should be `this.entity.model.meshInstances[0]...`

Reply:
fixing, thanks